addons/sale/models/payment_transaction.py

```diff
@@ -49,54 +49,45 @@ def _set_pending(self, state_message=None, **kwargs):
+        """ Update the sale order based on the amount of the transaction.
 
-        Confirm the sales orders only if the transaction amount is equal to the total amount of the
-        sales orders. Neither partial payments nor grouped payments (paying multiple sales orders in
-        one transaction) are not supported.
+        Confirm the sales orders only if the transaction amount (or the sum of the partial
+        transaction amounts) is equal to the total amount of the sales orders. Grouped payments
+        (paying multiple sales orders in one transaction) are not supported.
 
-        :return: The confirmed sales orders.
-        :rtype: a `sale.order` recordset
+        :return: None.
         """
         confirmed_orders = self.env['sale.order']
         for tx in self:
-            # We only support the flow where exactly one quotation is linked to a transaction and
-            # vice versa.
+            # We only support the flow where exactly one quotation is linked to a transaction.
             if len(tx.sale_order_ids) == 1:
                 quotation = tx.sale_order_ids.filtered(lambda so: so.state in ('draft', 'sent'))
                 if quotation and len(quotation.transaction_ids.filtered(
                     lambda tx: tx.state in ('authorized', 'done')  # Only consider confirmed tx
-                )) == 1:
-                    # Check if the SO is fully paid
-                    if quotation.currency_id.compare_amounts(tx.amount, quotation.amount_total) == 0:
+                )) >= 1:
+                    comparison = quotation.currency_id.compare_amounts(
+                        quotation.amount_total, quotation.amount_paid
+                    )
+                    if comparison == 0:  # the SO is fully paid, confirm it
                         quotation.with_context(send_email=True).action_confirm()
                         confirmed_orders |= quotation
-                    else:
-                        _logger.warning(
-                            '<%(provider)s> transaction AMOUNT MISMATCH for order %(so_name)s '
-                            '(ID %(so_id)s): expected %(so_amount)s, got %(tx_amount)s', {
-                                'provider': tx.provider_code,
-                                'so_name': quotation.name,
-                                'so_id': quotation.id,
-                                'so_amount': format_amount(
-                                    quotation.env, quotation.amount_total, quotation.currency_id
-                                ),
-                                'tx_amount': format_amount(tx.env, tx.amount, tx.currency_id),
-                            },
-                        )
+                    elif comparison == 1:
+                        # some amount remain to be paid, inform the customer of the payment
+                        # nevertheless
+                        quotation._send_order_payment_executed_mail()
```

Comment:
No, this is done after, currently L90 and L108. You don't have to separate `confirmed` orders from other anymore =]